### PR TITLE
Better synopses in many packages for a prettier catalog

### PR DESCRIPTION
**Pull request by xymus**
This should fix/complete the synopses of my projects. The few cases left are where the doc is in the main module of a contrib without a readme, like with `header_keeper`. I'm not sure what to do with those yet, maybe simply removing the src folder.

I seizes this opportunity to improve the `extern_methods` example and create missing readme files. And also reorganized the mnit platform implementations. There is still similar work to do with apps and submodules names to clean in mnit, but it will be for a future PR.

### CODE REVIEW

File: examples/extern_methods.nit [hunk 1]
```diff
@@ -28,22 +31,27 @@ redef enum Int
...
-	# * the + operator, a method of Int
-	# * to_s, a method of all objects
-	# * String.to_cstring, a method of String to return an equivalent char*
+	# Print the result of adding `self` to `self.fib` with callbacks from C
+	#
+	# This method illustrates many forms of callbacks from C,
+	# but it would have been better implemented in Nit (as done by `bar`).
+	#
+	# It is recommended to avoid callbacks when possible, use each language according to their strengths.
+	# Nit is better than C to call methods manipulate objects.A
```
Comment (privat) [inline examples/extern_methods.nit]: `.A`?

File: examples/extern_methods.nit [hunk 2]
```diff
@@ -28,22 +31,27 @@ redef enum Int
...
+	# C has easy access to system function, native libraries and it does low-level stuff.
```
Comment (privat) [inline examples/extern_methods.nit]: function**s** ?

File: examples/extern_methods.nit [hunk 3]
```diff
@@ -54,16 +62,25 @@ redef enum Int
 		printf("from C: self + fib(self) = %s\n", c_string);
 	`}
 
-	# Equivalent to foo but written in pure Nit
+	# Equivalent to `foo` but written in pure Nit
 	fun bar do print "from Nit: self + fib(self) = {self+self.fib}"
 end
 
+redef class Float
+	# Arctangent of `self` and `other` using `atan2` from libmath
```
Comment (privat) [inline examples/extern_methods.nit]: the usual name for the C math library is `math.h`. So "using `atan2` from `math.h`"

File: lib/bucketed_game.nit [hunk 4]
```diff
@@ -14,12 +14,15 @@
...
 #
-# Allows for fast support of a large number of entities with rare actions,
-# such as a forest with many individual trees.
+# Provides basic game logic entities to manage a game where the logic is executed by turns:
+# `Game`, `GameTurn`, `GameEvent`, `Turnable`.
+# Also offers a bucket system to plan future events at a known number of turns in the future:
+# `Bucketable` and the services `act_next` and `act_in`.
+#
+# Efficiently support large number of entities with rare or sparse actions,
+# such as a forest with many individual trees growing slowly.
```
Comment (privat) [inline lib/bucketed_game.nit]: Hey! What about a forest simulator with trees, lumberjacks and termites?

File: lib/curl/curl.nit [hunk 5]
```diff
@@ -14,7 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Curl services: `CurlHTTPRequest` and `CurlMail`
+# Data transfer with URL syntax from libcurl
```
Comment (privat) [inline lib/curl/curl.nit]: Title gore? I cannot parse:

* Data transfer (with (URL syntax) from libcurl)
* (Data transfer with URL) (syntax from libcurl)
* (Data transfer with (URL syntax)) from libcurl

File: lib/egl.nit [hunk 6]
```diff
@@ -14,8 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# EGL is an interface between the rendering APIs OpenGL, OpenGL ES, etc.
-# and the native windowing system.
+# EGL, an interface between rendering APIs (OpenGL, OpenGL ES, etc.) and the native windowing system.
```
Comment (privat) [inline lib/egl.nit]: Superfluous project name repetition?

File: lib/bucketed_game.nit (hunk 4 above)
Comment (xymus) [inline lib/bucketed_game.nit]: That would be such a cool game!

File: lib/glesv2/examples/opengles2_hello_triangle.nit [hunk 7]
```diff
@@ -22,7 +22,7 @@ module opengles2_hello_triangle
 
 import glesv2
 import egl
-import mnit_linux::sdl
+import mnit::linux::sdl
```
Comment (privat) [inline lib/glesv2/examples/opengles2_hello_triangle.nit]: in fact, sdl is its own project, the import is bad. see #1667

File: lib/json/json.nit [hunk 8]
```diff
@@ -15,7 +15,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Provides two APIs to manipulate JSON strings.
+# APIs to manipulate JSON strings.
```
Comment (privat) [inline lib/json/json.nit]: s/API/Library/ (as an API is just the interface)

File: lib/curl/curl.nit (hunk 5 above)
Comment (xymus) [inline lib/curl/curl.nit]: I tried using the synopsis from http://curl.haxx.se/

File: lib/mpd.nit [hunk 9]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Client for a MPD server
+# Music Player Daemon client API
```
Comment (privat) [inline lib/mpd.nit]: s/API/Library/

File: lib/egl.nit (hunk 6 above)
Comment (xymus) [inline lib/egl.nit]: Maybe, but it's weird from inside the module...

File: lib/realtime.nit [hunk 10]
```diff
@@ -10,7 +10,7 @@
 # You  are  allowed  to  redistribute it and sell it, alone or is a part of
 # another product.
 
-# Provides the Clock utility class to keep time of real time flow
+# Keep time of the real wall clock time with `Clock`
```
Comment (privat) [inline lib/realtime.nit]: I think that the synopsis should be clear if this is a library, a tool (or both)

File: lib/sdl.nit [hunk 11]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# SDL display support (used in Linux for windows and inputes only)
+# SDL display support (used in Linux for windows and inputs only)
```
Comment (privat) [inline lib/sdl.nit]: Title gore? Cannot parse

**Comment by privat** [comment]
+1

**Comment by privat** [comment]
Achievement unlocked: Quis custodiet ipsos custodes?. You broke nitpick: http://gresil.org/jenkins/job/CI_github/3444/testReport/cmd/nit/errors/

File: contrib/physical_interface_for_mpd_on_rpi/physical_interface_for_mpd_on_rpi.nit [hunk 12]
```diff
@@ -14,10 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# This programs interprets the input of a physical interface thought the
-# GPIO pins of a Raspberry Pi to control an MPD server.
+# Backend to a hardware interface to control an MPD server from a Raspberry Pi
 #
-# It suppot two inputs: a play/pause button and a rotary encoder to adjust
+# It support two inputs: a play/pause button and a rotary encoder to adjust
```
Comment (R4PaSs) [inline contrib/physical_interface_for_mpd_on_rpi/physical_interface_for_mpd_on_rpi.nit]: :s/support/supports

File: examples/extern_methods.nit [hunk 13]
```diff
@@ -28,22 +31,27 @@ redef enum Int
...
-	# * the + operator, a method of Int
-	# * to_s, a method of all objects
-	# * String.to_cstring, a method of String to return an equivalent char*
+	# Print the result of adding `self` to `self.fib` with callbacks from C
+	#
+	# This method illustrates many forms of callbacks from C,
+	# but it would have been better implemented in Nit (as done by `bar`).
+	#
+	# It is recommended to avoid callbacks when possible, use each language according to their strengths.
+	# Nit is better than C to call methods manipulate objects.
```
Comment (R4PaSs) [inline examples/extern_methods.nit]: :s/methods manipulate/methods and manipulate/

**Comment by privat** [comment]
I think that maybe you should split the mnit merging issue with the documentation issue. Maybe this PR could contains documentation for all but mnit related things. And open a new one with the mnit things

**Comment by xymus** [comment]
Rerolled to fix typos and remove the mnit part.

**Comment by privat** [comment]
+1

**Comment by xymus** [comment]
Fix test.

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**